cpp/src/arrow/util/utf8_util_test.cc

```diff
@@ -562,5 +564,30 @@ TEST(UTF8Length, Basics) {
   ASSERT_EQ(length("\xf0\x9f\x99\x8c"), 1);
 }
 
+class UTF8ExtensionArrayTest : public ::testing::Test {
+ public:
+  static std::shared_ptr<Array> ExampleJson(
+      const std::shared_ptr<DataType>& storage_type) {
+    std::shared_ptr<Array> arr = ArrayFromJSON(storage_type, R"([
+    "null",
+    "1234",
+    "3.14159",
+    "true",
+    "false",
+    "\"a json string\"",
+    "[\"a\", \"json\", \"array\"]",
+    "{\"obj\": \"a simple json object\"}"
+   ])");
+    return ExtensionType::WrapArray(arrow::extension::json(storage_type), arr);
+  }
+};
+
+TEST_F(UTF8ExtensionArrayTest, JSONExtensionType) {
```

Comment:
IMHO, this should simply go in `json_test.cc`.

Reply:
Moved to `json_test.cc`.